### ISSUE TITLE
how to use a public jupyter notebook server with different notebook_dir?

### ISSUE THREAD

**Issue by huwenfeng** [jupyter/notebook#4920]
dear all,
I want to integrate notebook to my website. Is it possible to start a public notebook server for many notebook_dir? 
I just want to identify them by  URL , with specified directory path parameters, but don'tstart new notebook server?

thanks very  much!

**Comment by 00Kai0** [comment]
Hi @huwenfeng , maybe you can have a look about [jupyterhub](https://github.com/jupyterhub/jupyterhub).

**Comment by bollwyvl** [comment]
Right, as a general principle, it is NEVER SAFE to run a public (as in the
whole world) notebook server. JupyterHub, as documented in
https://zero-to-jupyterhub.readthedocs.io
<https://zero-to-jupyterhub.readthedocs.io/en/latest/> is the minimum
acceptable protection for remote-code-execution-as-a-service. It expects
kubernetes, either on an in-house cluster, or from a commodity cloud vendor.

If you PERSONALLY KNOW the people that use the server, e.g. your coworkers
or research group, AND it is behind SSL (or better still a VPN) it CAN be
okay to run https://tljh.jupyter.org. It can be used on your own (modern
debian) server, Digital Ocean, and other low-cost hosts.

As an absolute last resort, for example if all of the users already have
shell access over SSH on the server, it can be fine to run notebook, but
even then you would want each of them to run their own notebook server
process, as the notebook server is not intended to be used by multiple
users.

If you just want to render notebooks, you can also run a local instance of
nbviewer.org, but even then you need to be careful.
